packages/nodes-base/nodes/Gitlab/Gitlab.node.ts

```diff
@@ -971,6 +1020,267 @@ export class Gitlab implements INodeType {
+				type: 'boolean',
+				default: true,
+				displayOptions: {
+					show: {
+						operation: ['get'],
+						resource: ['file'],
+					},
+				},
+				description:
+					'Whether to set the data of the file as binary property instead of returning the raw API response',
+			},
+			{
+				displayName: 'Binary Property',
+				name: 'binaryPropertyName',
+				type: 'string',
+				default: 'data',
+				required: true,
+				displayOptions: {
+					show: {
+						asBinaryProperty: [true],
+						operation: ['get'],
+						resource: ['file'],
+					},
+				},
+				placeholder: '',
+				description:
+					'Name of the binary property in which to save the binary data of the received file',
+			},
+			{
+				displayName: 'Additional Parameters',
+				name: 'additionalParameters',
+				placeholder: 'Add Parameter',
+				description: 'Additional fields to add',
+				type: 'collection',
+				default: {},
+				displayOptions: {
+					show: {
+						operation: ['get'],
+						resource: ['file'],
+					},
+				},
+				options: [
+					{
+						displayName: 'Reference',
+						name: 'reference',
+						type: 'string',
+						default: '',
+						placeholder: 'master',
+						description:
+							'The name of the commit/branch/tag. Default: the repository’s default branch (usually master).',
+					},
+				],
+			},
+
+			// ----------------------------------
+			//         file:create/edit
+			// ----------------------------------
+			{
+				displayName: 'Binary Data',
+				name: 'binaryData',
+				type: 'boolean',
+				default: false,
+				required: true,
+				displayOptions: {
+					show: {
+						operation: ['create', 'edit'],
+						resource: ['file'],
+					},
+				},
+				description: 'Whether the data to upload should be taken from binary field',
+			},
+			{
+				displayName: 'File Content',
+				name: 'fileContent',
+				type: 'string',
+				default: '',
+				required: true,
+				displayOptions: {
+					show: {
+						binaryData: [false],
+						operation: ['create', 'edit'],
+						resource: ['file'],
+					},
+				},
+				placeholder: '',
+				description: 'The text content of the file',
+			},
+			{
+				displayName: 'Binary Property',
+				name: 'binaryPropertyName',
+				type: 'string',
+				default: 'data',
+				required: true,
+				displayOptions: {
+					show: {
+						binaryData: [true],
+						operation: ['create', 'edit'],
+						resource: ['file'],
+					},
+				},
+				placeholder: '',
+				description: 'Name of the binary property which contains the data for the file',
+			},
+			{
+				displayName: 'Commit Message',
+				name: 'commitMessage',
+				type: 'string',
+				default: '',
+				required: true,
+				displayOptions: {
+					show: {
+						operation: ['create', 'delete', 'edit'],
+						resource: ['file'],
+					},
+				},
+			},
+			{
+				displayName: 'Branch',
+				name: 'branch',
+				type: 'string',
+				default: '',
+				description: 'Name of the new branch to create. The commit is added to this branch.',
```

Comment:
With the Github node the branch is an additional option and the API will just use the default repo branch, Does Gitlab not do the same thing?

Reply:
No, GitLab doesn't do the same thing. As I mentioned in my very first comment in this PR I wanted to make the GitLab node work like the GitHub node. I only made a change here because the GitLab API works different for this part.

Reply:
@svzi perfect, I missed that bit :)